### PR TITLE
Fully regenerate CodepointWidthDetector from Unicode 13.0

### PR DESCRIPTION
**Pull request by DHowett**
This commit also adds an override UCD and migrates all of the overrides
from GetQuickCharWidth into it.

GetQuickCharWidth
-----------------

The removal of overrides from GQCW reduces the number of comparisons
required for looking up a single character's width from 41 (32
individual ranged comparisons from GQCW + 8+1 from the binary search in
CPWD) to 11 (2 from GQCW, 8+1 from CPWD).

GQCW also incorrectly marked 67 reserved codepoints as `Wide` when they
should have been `Narrow`.

The codepoints whose definitions have changed from `Wide` to `Narrow` are:

```
2E9A 2EF4 2EF5 2EF6 2EF7 2EF8 2EF9 2EFA 2EFB 2EFC 2EFD 2EFE 2EFF 2FD6
2FD7 2FD8 2FD9 2FDA 2FDB 2FDC 2FDD 2FDE 2FDF 2FE0 2FE1 2FE2 2FE3 2FE4
2FE5 2FE6 2FE7 2FE8 2FE9 2FEA 2FEB 2FEC 2FED 2FEE 2FEF 2FFC 2FFD 2FFE
2FFF 31E4 31E5 31E6 31E7 31E8 31E9 31EA 31EB 31EC 31ED 31EE 31EF 321F
A48D A48E A48F FE1A FE1B FE1C FE1D FE1E FE1F FE53 FE67
```

All of them are reserved, but those reserved regions are marked as narrow
in the UCD.

This change also offers us the chance to document exactly why we're
overriding a specific character range. Comments from the override
document will be copied to the generated CPWD table.

New in Unicode 13.0
------------------

Some widths have changed due to previously-reserved characters becoming
_used_ such as U+32FF SQUARE ERA NAME REIWA, the Tangut components
756-768, the entire Khitan Small Script character set, and the Tangut
Ideographs.

A number of the changes in this diff are due to better/worse comment
tracking and the removal of the Emoji/EPres comments. The script once
mistakenly applied comments to packed regions (and it has been updated
to not do so.)

Validation
----------

I build a test application that compared codepoints 0-FFFF for GQCW
against their new registered widths.

### CODE REVIEW

**Review by zadjii-msft — APPROVED** [review]
Sure, seems like a good idea to me

File: src/types/unicode_width_overrides.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8" standalone="yes"?>
+<ucd xmlns="http://www.unicode.org/ns/2003/ucd/1.0">
+   <repertoire>
```
Comment (miniksa) [inline src/types/unicode_width_overrides.xml:3]: Talk about a tough tag name to spell correctly...

File: src/types/convert.cpp [hunk 2]
```diff
@@ -354,179 +354,16 @@ std::deque<std::unique_ptr<KeyEvent>> SynthesizeNumpadEvents(const wchar_t wch,
 // May-01-2019 MiNiksa  Forced lookup-via-renderer for retroactively recategorized emoji
 //                      that used to be narrow but now might be wide. (approx x2194-x2b55, not inclusive)
 //                      Also forced block characters segment (x2580-x259F) to narrow
+// Oct-25-2020 DuHowett Replaced the entire table with a set of overrides that get built into
```
Comment (miniksa) [inline src/types/convert.cpp:357]: I like this plan, overrides declarative not in here.

**Review by miniksa — APPROVED** [review]

**Comment by plastikfan** [comment]
If anyone knows, please can you let me know which version of Unicode WIndows Terminal currently supports, thanks.

**Comment by DHowett** [comment]
@plastikfan do you have any reason to believe that it is not, in fact, 13.0?

**Comment by plastikfan** [comment]
No not all, but if thats the answer then great. Please confirm

**Comment by DHowett** [comment]
In the issue you linked to this pull request, there is a discussion about the version of Unicode that is supported _by the Windows Console_. The version that Windows Terminal (and OpenConsole) supports doesn't mean _anything_ for FluentTerm, because they are not using Windows Terminal (or OpenConsole)... so you're somewhat barking up the wrong tree in asking.

However: the inbox Windows console is updated from this repository, and this change made it into builds numbered 21xxx. Before that, we supported some smattering of Unicode versions 9, 10, and 11.

**Comment by plastikfan** [comment]
Sorry, I think you mis-intepret me. All I'm trying to do is assist in researching why emoji display is not working properly in Fluent terminal. I was asked to find out what version of Unicode Windows terminal is currently using (so that the correct so called unicode addon can be loaded into xterm) and with your info, this is done. Thank you for responding.

**Comment by DHowett** [comment]
If xterm.js supports Unicode 13, and Fluent Terminal is using the traditional in-box Windows console, **it will disagree about the size of a character with xterm.js** which will cause visual artifacts.

**Pull request merged**